### PR TITLE
Refactor image resolvers, and minor fixes found while adding the CentOS resolver

### PR DESCRIPTION
**Pull request by mikalstill**

### CODE REVIEW

File: shakenfist/baseobject.py [hunk 1]
```diff
@@ -100,7 +99,18 @@ def state(self, new_value):
...
+                return
+
             # Ensure state change is valid
+            if not self.state_targets:
+                raise exceptions.NoStateTransitionsDefined(
+                    self.object_type)
+
+            LOG.with_field('original state', orig).with_field(
+                'new state', new_value).with_field(
+                    'targets', self.state_targets).debug('Verifying state transition')
```
Comment (mandoonandy) [inline shakenfist/baseobject.py:112]: This could be less verbose using `with_fields()`?

File: shakenfist/image_resolver/debian.py [hunk 2]
```diff
@@ -44,14 +41,18 @@ def resolve(name):
...
         'name': name,
         'url': url,
         'checksum': checksum
     }).info('Image resolved')
-    return (url, checksum)
+
+    if checksum:
+        return (url, checksum)
+    else:
+        return (url, None)
```
Comment (mandoonandy) [inline shakenfist/image_resolver/debian.py:58]: Would this be better here?
```
return (url, checksum if checksum else None)
```

File: shakenfist/image_resolver/ubuntu.py [hunk 3]
```diff
@@ -56,12 +52,16 @@ def resolve(name):
...
         'name': name,
         'url': url,
         'checksum': checksum
     }).info('Image resolved')
-    return (url, checksum)
+
+    if checksum:
+        return (url, checksum)
+    else:
+        return (url, None)
```
Comment (mandoonandy) [inline shakenfist/image_resolver/ubuntu.py:67]: as above, using `if`

File: shakenfist/images.py [hunk 4]
```diff
@@ -63,8 +56,8 @@ def __init__(self, static_values):
 
     @classmethod
     def new(cls, url, checksum=None):
-        # Handle URL shortcut with built-in resolvers
-        url, resolver_checksum = Image._resolve(url)
+        # Handle URL shortcut with built-in , '
```
Comment (mandoonandy) [inline shakenfist/images.py:59]: missed enter, hit `'`?

**Review by mandoonandy — CHANGES_REQUESTED** [review]
Couple of tiny comments to prove I looked at the code.

File: shakenfist/image_resolver/debian.py (hunk 2 above)
Comment (mikalstill) [reply shakenfist/image_resolver/debian.py:58]: Done.

File: shakenfist/images.py (hunk 4 above)
Comment (mikalstill) [reply shakenfist/images.py:59]: Great moments in search and replace.

**Review by mandoonandy — APPROVED** [review]
LGTM

**Pull request merged**